google/cloud/bigtable/data/_metrics/data_model.py

```diff
@@ -0,0 +1,514 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+from typing import Callable, Any, Generator, Tuple, cast, TYPE_CHECKING
+
+import datetime
+import time
+import os
+import re
+import logging
+
+from enum import Enum
+from functools import lru_cache
+from dataclasses import dataclass
+from dataclasses import field
+from grpc import StatusCode
+
+import google.cloud.bigtable.data.exceptions as bt_exceptions
+
+if TYPE_CHECKING:
+    from google.cloud.bigtable.data._metrics.handlers._base import MetricsHandler
+
+
+# by default, exceptions in the metrics system are logged,
+# but enabling this flag causes them to be raised instead
+ALLOW_METRIC_EXCEPTIONS = os.getenv("BIGTABLE_METRICS_EXCEPTIONS", False)
```

Comment:
Yeah, this is for raising an exception if anything goes wrong in the metrics system, instead of just logging. Could be useful for debugging, but I'm also fine with just removing it before merging

Reply:
Sure, done